node/actors/control_api.go

```diff
@@ -181,16 +181,108 @@ func (api *ControlAPI) subscribe() error {
 }
 
 func (api *ControlAPI) handleAuction(m *nats.Msg) {
-	// TODO
+	req := new(nodecontrol.AuctionRequestJson)
+	err := json.Unmarshal(m.Data, req)
+	if err != nil {
+		api.logger.Error("Failed to unmarshal auction request", slog.Any("error", err))
+		respondEnvelope(m, AuctionResponseType, 500, nil, fmt.Sprintf("failed to unmarshal auction request: %s", err))
+		return
+	}
+
+	ctx := context.Background()
+	_, agentSuper, err := api.self.ActorSystem().ActorOf(ctx, AgentSupervisorActorName)
+	if err != nil {
+		api.logger.Error("Failed to locate agent supervisor actor", slog.Any("error", err))
+		respondEnvelope(m, AuctionResponseType, 500, nil, fmt.Sprintf("failed to locate agent supervisor actor: %s", err))
+		return
+	}
+
+	askResp, err := api.self.Ask(ctx, agentSuper, auctionRequestToProto(req))
```

Comment:
the auction request may not need to go to the agent supervisor. Can the auction be satisfied with just node config and whatever info the api server has on hand?

Reply:
yea, this was my confusion.  Seems like this (and the few other node specific requests) Ask needs to go to the node supervisor I would think?

Reply:
There is no node supervisor ;) We should be able to make a regular function call to get the data necessary to respond to these queries.